solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/provisioning/SolaceQueueProvisioner.java

```diff
@@ -193,15 +191,28 @@ private Queue provisionErrorQueue(String errorQueueName, SolaceConsumerPropertie
 		return provisionQueue(errorQueueName, true, endpointProperties, properties.isProvisionErrorQueue(), "Error Queue");
 	}
 
-	public void addSubscriptionToQueue(Queue queue, String topicName, SolaceCommonProperties properties) {
+	public void addSubscriptionToQueue(Queue queue, String topicName, SolaceCommonProperties properties, boolean isDestinationSubscription) {
 		logger.info(String.format("Subscribing queue %s to topic %s", queue.getName(), topicName));
 
-		if (queue.isDurable() && !properties.isProvisionSubscriptionsToDurableQueue()) {
+		if (!isDestinationSubscription && queue.isDurable() && !properties.isProvisionSubscriptionsToDurableQueue()) {
 			logger.warn(String.format("Adding subscriptions to durable queues was disabled, queue %s will not be subscribed to topic %s",
 					queue.getName(), topicName));
 			return;
 		}
 
+		//This condition is for backward compatibility and will be removed when the deprecated property provisionSubscriptionsToDurableQueue is removed
+		if (isDestinationSubscription && queue.isDurable() && properties.isAddDestinationAsSubscriptionToQueue() && !properties.isProvisionSubscriptionsToDurableQueue()) {
+			logger.warn(String.format("Adding subscription subscription to durable was disabled, queue %s will not be subscribed to topic %s",
+					queue.getName(), topicName));
+			return;
+		}
+
+		if (isDestinationSubscription && !properties.isAddDestinationAsSubscriptionToQueue()) {
+			logger.warn(String.format("Adding destination subscription was disabled, queue %s will not be subscribed to topic %s",
+					queue.getName(), topicName));
+			return;
+		}
+
```

Comment:
I think this could be simplified to be something like:
```
if (isDestinationSubscription && (
		!properties.isAddDestinationAsSubscriptionToQueue() ||
		(queue.isDurable() && !properties.isProvisionSubscriptionsToDurableQueue()))) {
	logger.warn(String.format("Adding destination as subscription was disabled, queue %s will not be subscribed to topic %s",
		queue.getName(), topicName));
	return;
}
```

Reply:
You are right that we can save a few lines of code, however, I think keeping the conditions as they are right now is easier to reason about. Also, when comes time to remove the deprecated parameter, we can quickly and simply delete that one if statement. 
I took the opportunity to improve the log messages. We log a slightly different message based on the context which is another argument to keep the current code.